addon/index.js

```diff
@@ -25,19 +24,22 @@ const mod = new PageMod({
   onAttach: setupApp
 });
 
+if (!store.experiments) store.experiments = {};
+if (!store.availableExperiments) store.availableExperiments = [];
+
 require('sdk/system/unload').when(function(reason) {
   if (reason === 'uninstall') {
     app.send('addon-self:uninstalled');
     if (store.experiments) {
-      uninstall(store.experiments);
+      Object.keys(store.experiments).forEach(e => uninstall({addon_id: e}));
```

Comment:
Since you don't care about iterating experiments in order, and because the keys are the addon IDs (which is what you're after), you can just get the keys via `for in`:
```
for (e in store.experiments) {
  uninstall({addon_id: e});
}
```
MDN docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in